### PR TITLE
WIP Cleanup and add AmazonS3Cache to build

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/JimBobSquarePants/ImageProcessor/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practise as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Adds AmazonS3Cache to the solution and packages.

@AndyFelton If you could review this please and also provide descriptions for each of the config properties that would be great!

### CODE REVIEW

File: src/ImageProcessor.Web.Plugins.AmazonS3Cache/AmazonS3Cache.cs [hunk 1]
```diff
@@ -123,13 +120,13 @@ public class AmazonS3Cache : ImageCacheBase
         public AmazonS3Cache(string requestPath, string fullPath, string querystring)
             : base(requestPath, fullPath, querystring)
         {
-            awsAccessKey = Settings.ContainsKey("AwsAccessKey") ? Settings["AwsAccessKey"] : string.Empty;
-            awsSecretKey = Settings.ContainsKey("AwsSecretKey") ? Settings["AwsSecretKey"] : string.Empty;
+            this.awsAccessKey = this.Settings.ContainsKey("AwsAccessKey") ? this.Settings["AwsAccessKey"] : string.Empty;
```
Comment (JimBobSquarePants) [inline src/ImageProcessor.Web.Plugins.AmazonS3Cache/AmazonS3Cache.cs]: These should be static.

**Comment by AndyFelton** [comment]
@JimBobSquarePants just checked the code and it all looks OK to me. Will provide the meanings for the config properties shortly.

Sorry for the delays on this - been busy recently

**Comment by AndyFelton** [comment]
@JimBobSquarePants 

Config properties as follows:
- _AwsBucketName_ - S3 Bucket Name
- _AwsCacheKeyPrefix_ - S3 Object Key Prefix
- _RegionEndpoint_ - AWS Region Endpoint for writing processed images back too (us-east-1/us-west-1/ap-southeast-2)
- _SourceRegionEndpoint_ - AWS Region Endpoint for source images (us-east-1/us-west-1/ap-southeast-2)
- _CachedCDNRoot_ - Full CDN root url e.g http://123456.vo.msecnd.net/ 
- _CloudSourceURI_ - URI for source images
- _CachedCDNTimeout_ - Optional setting for a timeout limit in milliseconds when attempting to communicate with the CDN url.
- _StreamCachedImage_ -  Optional settings facilitate streaming of the blob resource directly instead of a redirect. This is beneficial for CDN purposes but caution should be taken if not used with a CDN as it will add quite a bit of overhead to the site.

### Access keys
How access keys are specified depends on how you host your project. The keys will already exist if deploying inside an EC2 instance (specified via environment variables during deployment and creation of infrastructure). If you aren't using EC2 or ElasticBeanstalk to access generated keys the following config can also be added
- _AwsAccessKey_ - IAM access key
- _AwsSecretKey_ - IAM secret

**Comment by JimBobSquarePants** [comment]
Wizard. Thanks! I'm gonna have another pass of the code just to make sure it's efficient then I'll publish something.

**Comment by andyfelton2** [comment]
@JimBobSquarePants I know you're really busy at the moment but wondering when you could do a release including this pull request?

Thanks

**Comment by JimBobSquarePants** [comment]
@AndyFelton Gonna try and get a release out in the next few days.

**Comment by AndyFelton** [comment]
@JimBobSquarePants Sorry for the delay had a busy few days - fired up the test and configured it and all seems to be good. As always thanks for looking at this!

**Pull request merged**